vm/src/obj/objmodule.rs

```diff
@@ -18,23 +16,21 @@ impl PyValue for PyModule {
 
 impl PyModuleRef {
     fn dir(self: PyModuleRef, vm: &VirtualMachine) -> PyResult {
-        let keys = self
-            .dict
-            .get_key_value_pairs()
-            .iter()
-            .map(|(k, _v)| k.clone())
-            .collect();
-        Ok(vm.ctx.new_list(keys))
-    }
-
-    fn set_attr(self, attr: PyStringRef, value: PyObjectRef, vm: &VirtualMachine) {
-        self.dict.set_item(&vm.ctx, &attr.value, value)
+        if let Some(dict) = &self.into_object().dict {
+            let keys = dict
+                .get_key_value_pairs()
+                .iter()
+                .map(|(k, _v)| k.clone())
+                .collect();
+            Ok(vm.ctx.new_list(keys))
+        } else {
+            panic!("Modules should definitely have a dict.");
+        }
     }
 }
 
 pub fn init(context: &PyContext) {
     extend_class!(&context, &context.module_type, {
         "__dir__" => context.new_rustfunc(PyModuleRef::dir),
-        "__setattr__" => context.new_rustfunc(PyModuleRef::set_attr)
```

Comment:
Why should a module not have setattr?

Reply:
Because its setattr is now the same as the one it will inherit from object anyway.